rules/caught-error-name.js

```diff
@@ -0,0 +1,25 @@
+'use strict';
+
+module.exports = function (context) {
+	var opts = context.options[0];
+	var name = (opts && opts.name) || 'err';
+
+	var stack = [];
+
+	return {
+		'CatchClause': function (node) {
+			if (stack.length === 1) {
+				stack[0] = true;
+			}
+			stack.push(stack.length > 0 || node.param.name === name);
+		},
+		'CatchClause:exit': function (node) {
+			if (!stack.pop()) {
+				context.report({
+					node: node.param,
+					message: 'The catch clause parameter should be named ' + name
+				});
+			}
+		}
+	};
+};
```

Comment:
Can you provide a options schema to validate the option: https://github.com/sindresorhus/eslint-plugin-xo/blob/1c287ab3e0a0c9a57a4194eac6b17d8bbeb7fda6/rules/filename-case.js#L49-L60